src/kernel/boot.c

```diff
@@ -839,31 +834,80 @@ BOOT_CODE static bool_t check_available_memory(word_t n_available,
         /* Available regions must be sane */
         if (r->start > r->end) {
             printf("ERROR: memory region %"SEL4_PRIu_word" has start > end\n", i);
-            return false;
+            return 0;
         }
 
         /* Available regions can't be empty. */
         if (r->start == r->end) {
             printf("ERROR: memory region %"SEL4_PRIu_word" empty\n", i);
-            return false;
+            return 0;
         }
 
         /* Regions must be ordered and must not overlap. Regions are [start..end),
-           so the == case is fine. Directly adjacent regions are allowed. */
+         * so the '=='-case is fine. Directly adjacent regions are allowed.
+         */
         if ((i > 0) && (r->start < available[i - 1].end)) {
             printf("ERROR: memory region %d in wrong order\n", (int)i);
-            return false;
+            return 0;
+        }
+
+        /* Sanitize the available memory region, only such memory can be used
+         * for kernel objects that is visible in the kernel window. On 32-bit
+         * architectures loosing some memory here is not uncommon, because the
+         * 32-bit virtual address space is too small to support a kernel window
+         * that is big enough. Memory outside of he kernel window is not list,
+         * it is made available as device untyped memory.
+         */
```

Comment:
This should be rephrased a bit. Maybe:

> Sanitize the available memory region: only memory in the kernel window can be used for kernel objects. On 32-bit architectures it is not uncommon for provided regions for provided regions to be only partially in the kernel window, because the available address space is small. Memory outside of the kernel window is made available as device untyped memory.

Reply:
done